source/base/polynomial.cc

```diff
@@ -1228,6 +1234,239 @@ namespace Polynomials
     return basis;
   }
 
+
+// ------------------ HermiteLikeInterpolation --------------- //
+  namespace
+  {
+    // Finds the zero position x_star such that the mass matrix entry (0,1)
+    // with the Hermite polynomials evaluates to zero. The function has
+    // originally been derived by a secant method for the integral entry
+    // l_0(x) * l_1(x) but we only need to do one iteration because the zero
+    // x_star is linear in the integral value.
+    double find_variable_support_point (const Vector<double> &jacobi_roots)
+    {
+      // Initial guess for the support point position values: The zero turns
+      // out to be between zero and the first root of the Jacobi polynomial,
+      // but the algorithm is agnostic about that, so simply choose two points
+      // that are sufficiently far apart.
+      double left_guess = 0;
+      double right_guess = 0.5;
+      const unsigned int degree = jacobi_roots.size() + 3;
+
+      // compute the integrals of the product of l_0(x) * l_1(x)
+      // l_0(x) = (x-x_zero)*(x-eigval(0))*...*(x-eigval(degree-3))*(x-1)*(x-1)
+      // l_1(x) = (x-0)*(x-eigval(0))*...*(x-eigval(degree-3))*(x-1)*(x-1)
```

Comment:
The last term in the expanded product should still be `(x-eigval(degree-4))` instead of `(x-eigval(degree-3))` to match the code below. Also the name of the vector changed to `jacobi_roots` and `x_zero` should be `x_star`  in the end but is `left_guess` or `right_guess` below.

Reply:
Ha, thanks for reading this more carefully than I did when revising it... 👍